### PR TITLE
AES-256 encryption using passphrase for permissions group id generation

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate

### CODE REVIEW

File: app/crypto.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+func passphraseToKey32AndIV16(passphrase []byte) ([]byte, []byte) {
+	keyAndIV := make([]byte, 48)
+	hashed := []byte{}
+	for i := 0; i < 3; i++ {
+		hashed = hash(hashed, passphrase)
+		copy(keyAndIV[i*16:], hashed)
+	}
+	return keyAndIV[:32], keyAndIV[32:]
```
Comment (jh-bate) [inline app/crypto.go]: another dumb question: so in order to guarantee we are in fact using the AES256 for encryption do you feel we assert well enough in the tests that we are getting 32 bytes here?

> NewCipher creates and returns a new cipher.Block. The key argument should be the AES key, either 16, 24, or 32 bytes to select AES-128, AES-192, or AES-256.

File: app/crypto.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ */
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/md5"
+)
+
+func EncryptWithAES256UsingPassphrase(bytes []byte, passphrase []byte) ([]byte, error) {
+	if len(bytes) == 0 {
```
Comment (jh-bate) [inline app/crypto.go]: after discussion with @darinkrauss the bytes being short is not an issue and also I  understand this is a utility that the passphrase would be coming from config and if we wanted to restrict we would place the check elsewhere

File: app/crypto.go (hunk 1 above)
Comment (jh-bate) [inline app/crypto.go]: sounds good to me @darinkrauss after discussion

**Comment by darinkrauss** [comment]
[<0:ux 5:dev 2:tst> [platform] Delete account and all related data](https://trello.com/c/ER64SIAH/8-0-ux-5-dev-2-tst-platform-delete-account-and-all-related-data)

**Pull request merged**